src/commands/watch/buildifier.rs

```diff
@@ -0,0 +1,49 @@
+use std::path::PathBuf;
+use uuid::Uuid;
+
+use super::uppificator::Pause;
+
+pub(crate) struct Buildifier {
+    pub spin_bin: PathBuf,
+    pub manifest: PathBuf,
+    pub clear_screen: bool,
+    pub watched_changes: tokio::sync::watch::Receiver<Uuid>, // TODO: refine which component(s) a change affects
```

Comment:
We use the `watch` channel because the receivers don't actually care which file has changed, and they don't need to respond to every change: they care only only that _something_ in their area of interest has changed since the last time they checked.  "Has something changed" is the exact use case for the `watch` channel, which does not transmit a sequence of values but holds just one mutable value; its `receive` returns when the value is different from the last `receive`.  This saves us from having to receive our way through a queue of multiple changes.

Using a `Uuid` means we don't need to worry about concocting a unique "what changed" message.  For example, if we sent the filename, then we couldn't use `watch` channels because if `fie.txt` changed (causing a reload) and then `fie.txt` changed again, that would _not_ be a change to the value in the channel.  `Uuid` guarantees that a new value will be generated for every debounced change set.

Reply:
Thanks for the education 😄